### PR TITLE
Test messages - FIXED

### LINKED ISSUE #84: Add tests for each messages class
As seen in the other test files
Test each class:
Messages
MessageBoard
MessageTags

add functions to make sure the models are correct.

### PR DESCRIPTION
**Pull request by shakedhagag**
## Tests for each Messages class:

This PR contains tests for each of the messages model class:
1. MessageBoards
2. Messages
3. MessageTags

This is the finished PR with everything running and working. 
Closes #84

### CODE REVIEW

**Review by DanielItkes — APPROVED** [review]

**Comment by DanielItkes** [comment]
Perfect.

File: AcadeMeData/test_messages.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    message = Messages(msgID=msgID, userID=user, text=text)
+    message.save()
+    return message
+
+
+@pytest.mark.django_db
+class TestMessagesModel:
+
+    def test_get_msg(self, generate_message):
+        msg = generate_message
```
Comment (Yarboa) [inline AcadeMeData/test_messages.py:26]: Please delete it,
the method already called in function declaration
Ho can you search a message? what is the key? 

So calling the fixture. maybe with id will be easier
assume your call to 
def test_get_msg(self, generate_message, msgID=3):

then you can search on line 27
msg_test = Messages.get_msg_by_id(msgID)

File: AcadeMeData/test_messages.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+def user_example():
+    user_data = {'username': "username2212", 'password': "password", 'email': "user@example.com",
+                 'university': "RU",
+                 'degree': "CS"}
+    user = User.create_user(*user_data)
+    return user
+
+
+@pytest.fixture
+def generate_message(user_example, msgID=1, text='This is a test message yo'):
```
Comment (Yarboa) [inline AcadeMeData/test_messages.py:15]: this fixture can call user_example fixture
please remove line 16

File: AcadeMeData/test_messages.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def test_get_msg(self, generate_message):
+        msg = generate_message
+        msg_test = Messages.get_msg_by_id(1)
+        assert msg_test == msg
+        assert isinstance(msg_test, Messages)
+
+    def test_create_msg(self, generate_message, user_example, msgID=9, text='I am testinggggg'):
+        user = user_example
+        msg = Messages.create_message(msgID, user, text)
```
Comment (Yarboa) [inline AcadeMeData/test_messages.py:33]: Please see test comments above 
This test covers  also the test above

File: AcadeMeData/test_messages.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        msg_test = generate_message
```
Comment (Yarboa) [inline AcadeMeData/test_messages.py:34]: Please delete line 34

File: AcadeMeData/test_msgboard.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+def generate_msgboard(id=1):
+    msgboard = MessageBoards(id=id, courseName="Linear Algebra")
+    msgboard.save()
+    return msgboard
+
+
+@pytest.mark.django_db
+class TestMessageBoardModel:
+    def test_create_msgboard(self, generate_msgboard):
+        board = generate_msgboard
```
Comment (Yarboa) [inline AcadeMeData/test_msgboard.py:15]: Please delete this line and thing how you could receive the board 
you already called the fixture in method declaration

File: AcadeMeData/test_messages.py (hunk 1 above)
Comment (shakedhagag) [reply AcadeMeData/test_messages.py:26]: Yariv I really dont understand you. 
The whole purpose of the test is to check if 2 instances of messages are the same or not
msg = generate_message ---> a variable which holds the fixture
msg_test is a variable that uses get msg by id function to hold a certain message
then after having these 2 vars I assert whether or not they are the same and if the message that came from the function get_msg is an instance of Messages class
BTW this code haven't been changed and you already approved it so I really dont understand the comments about an already seen and checked code.
I dont see nothing thats need to be changed here. 
(Some will say msg = generate_msg is not needed but its a matter of style..)

File: AcadeMeData/test_messages.py (hunk 2 above)
Comment (shakedhagag) [reply AcadeMeData/test_messages.py:15]: Again its as matter of style, nothing here isnt correct.
putting the fixture user_example in a variable user is the way I chose to do it
Even the space complexity (which is the only thing that comes to mind) is negligible!

File: AcadeMeData/test_messages.py (hunk 3 above)
Comment (shakedhagag) [reply AcadeMeData/test_messages.py:33]: please read the comments above

File: AcadeMeData/test_messages.py (hunk 4 above)
Comment (shakedhagag) [reply AcadeMeData/test_messages.py:34]: No need to imho

File: AcadeMeData/test_msgboard.py (hunk 5 above)
Comment (shakedhagag) [reply AcadeMeData/test_msgboard.py:15]: again...

File: AcadeMeData/test_msgtags.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    user_data = {'username': "username22123", 'password': "password", 'email': "user@example.com",
+                 'university': "RU",
+                 'degree': "CS"}
+    user = User.create_user(*user_data)
+    return user
+
+
+@pytest.fixture
+def generate_message(user_example, msgID=10, text='This is a test message yo'):
+    user = user_example
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:16]: no need 
search for user_example. the fixture return user, 
you can call it like that user_example it is holding it for you as User

File: AcadeMeData/test_msgtags.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    message = Messages(msgID=msgID, userID=user, text=text)
+    message.save()
+    return message
+
+
+@pytest.fixture
+def generate_msgtags(user_example, generate_message, id=1):
+    msg = generate_message
+    user = user_example
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:25]: see my comments above the fixture holds for you the return values

File: AcadeMeData/test_msgtags.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    tag = MessageTags.create_msgtag(id, msg, user)
+    return tag
+
+
+@pytest.mark.django_db
+class TestMessageTagsModel:
+    def test_get_msg_tag(self, generate_msgtags):
+        tag_test = MessageTags.get_msg_tag(1)
+        assert generate_msgtags == tag_test
```
Comment (Yarboa) [inline AcadeMeData/test_msgtags.py:34]: why do you understand it here, and in the above?

File: AcadeMeData/test_msgtags.py (hunk 6 above)
Comment (shakedhagag) [reply AcadeMeData/test_msgtags.py]: You're right 
Havent noticed in this file
FIXED

File: AcadeMeData/test_msgtags.py (hunk 7 above)
Comment (shakedhagag) [reply AcadeMeData/test_msgtags.py]: FIXED

File: AcadeMeData/test_msgtags.py (hunk 8 above)
Comment (shakedhagag) [reply AcadeMeData/test_msgtags.py:31]: FIXED

**Review by Yarboa — APPROVED** [review]
LGTM, Nice work

**Review by bregman-arie — APPROVED** [review]

**Pull request merged**